matrix-sdk/src/main/java/org/matrix/androidsdk/HomeserverConnectionConfig.java

```diff
@@ -0,0 +1,106 @@
+package org.matrix.androidsdk;
+
+import android.net.Uri;
+import android.util.Base64;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.matrix.androidsdk.rest.model.login.Credentials;
+
+import java.util.ArrayList;
+
+
+/**
+ * Represents how to connect to a specific Homeserver, may include credentials to use.
+ */
+public class HomeserverConnectionConfig {
+    private Uri mHsUri;
+    private byte[][] mAllowedFingerprints;
```

Comment:
Since we tend to store `HomeserverConnectionConfig` I would prefer that they remained fairly immutable for now. Although the fact you can set credentials might undermine that a bit.